src/input_context/preset.rs

```diff
@@ -0,0 +1,149 @@
+use bevy::prelude::*;
+
+use crate::{Input, InputBind, InputCondition, InputModifier, Negate, SwizzleAxis};
+
+pub trait BindPreset {
+    fn bindings(self) -> impl Iterator<Item = InputBind>;
```

Comment:
Functions should typically accept `Into`-likes, and never return `Into`-likes. Since, if `bindings` returns an `IntoIterator`, there's nothing that a caller can do with that function except for call `into_iter` on it. This step should be the implementor's responsibility.